Model/Order/PlaceOrder.php

```diff
@@ -247,10 +247,14 @@ public function authorizeAndPlace(string $publicOrderId, string $quoteMaskId): R
         $websiteId = (int)$this->storeManager->getStore()->getWebsiteId();
         $shopId = $this->config->getShopId($websiteId) ?? '';
 
-        try {
-            $quoteId = $this->maskedQuoteIdToQuoteId->execute($quoteMaskId);
-        } catch (NoSuchEntityException $exception) {
-            return $this->getValidationErrorResponse((string)__('Invalid quote mask ID "%1"', $quoteMaskId));
+        if (!is_numeric($quoteMaskId) && strlen($quoteMaskId) === 32) {
```

Comment:
This validation appears correct, just wondering where 32 came from & if this is reliable given quote functionality could potentially be customized (likely a very uncommon edge case)

Perhaps we could try unmasking the quote id, and assume it is unmasked if an exception is thrown? Though this could allow invalid quote ids to get past this validation

Reply:
I recommend verifying this validation with quotes from live merchants to ensure its reliable, apart from that lgtm!